### PR TITLE
add support for redis sentinel

### PR DESCRIPTION
**Pull request by accessd**
https://github.com/anycable/anycable/issues/2

### CODE REVIEW

**Comment by accessd** [comment]
Done! Could you enable Travis?
I have trouble with specs, there're random fails because sometimes `redis_sentinels` in config is empty:

```
#<Anycable::Config:0x007fe1bc93bc28
 @config_name="anycable",
 @connection_factory=Anycable::TestFactory,
 @debug=false,
 @redis_channel="anycable",
 @redis_sentinels=[],
 @redis_url="redis://localhost:6379/2",
 @rpc_host="localhost:50123">
```

File: spec/anycable/pubsub_spec.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+  context "with sentinel" do
+    it do
+      expect(Redis).to receive(:new).with(url: config.redis_url, sentinels: config.redis_sentinels)
+      subject
+    end
+  end
+
+  context "without sentinel" do
+    before do
```
Comment (palkan) [inline spec/anycable/pubsub_spec.rb]: You should add `after { Anycable.config.reload }` here to make sure that every test works with _non-dirty_ config.

**Comment by palkan** [comment]
> Could you enable Travis?

Please, try again. Should work now.

File: spec/anycable/pubsub_spec.rb (hunk 1 above)
Comment (accessd) [reply spec/anycable/pubsub_spec.rb]: With `after { Anycable.config.reload }` I have strange errors like:

```
subscriptions #unsubscribe invokes #unsubscribed for channel
     Failure/Error: subject { service.command(request) }

     GRPC::Unavailable:
       14:Connect Failed
     # ./vendor/bundle/gems/grpc-1.6.0-universal-darwin/src/ruby/lib/grpc/generic/active_call.rb:31:in `check_status'
     # ./vendor/bundle/gems/grpc-1.6.0-universal-darwin/src/ruby/lib/grpc/generic/active_call.rb:171:in `attach_status_results_and_complete_call'
     # ./vendor/bundle/gems/grpc-1.6.0-universal-darwin/src/ruby/lib/grpc/generic/active_call.rb:363:in `request_response'
     # ./vendor/bundle/gems/grpc-1.6.0-universal-darwin/src/ruby/lib/grpc/generic/client_stub.rb:152:in `request_response'
     # ./vendor/bundle/gems/grpc-1.6.0-universal-darwin/src/ruby/lib/grpc/generic/service.rb:170:in `block (3 levels) in rpc_stub_class'
     # ./spec/integrations/subscriptions_spec.rb:74:in `block (3 levels) in <top (required)>'
     # ./spec/integrations/subscriptions_spec.rb:83:in `block (4 levels) in <top (required)>'
     # ./spec/integrations/subscriptions_spec.rb:83:in `block (3 levels) in <top (required)>'
```

You can check it on Travis

Comment (palkan) [reply spec/anycable/pubsub_spec.rb]: There was a problem with config reloading (we set some configuration in our `spec_helper.rb` and it's not available after reload). I've fixed this in the upcoming version branch (`v0.5.0-dev`). Just rebase against it and everything should work.

And, please, update the changelog.

Comment (accessd) [reply spec/anycable/pubsub_spec.rb]: Done!

**Comment by palkan** [comment]
Thanks!

**Comment by tak1n** [comment]
Hey in case anybody implements the websocket server side support for redis sentinel, what is planned with the ruby sided broadcast support for redis sentinel?

I see it's merged but in branch 0.5.0-dev altough 0.6.0 is already released.

**Comment by palkan** [comment]
WS server support for sentinel hasn't been added yet, here is the issue: https://github.com/anycable/anycable-go/issues/38

**Comment by tak1n** [comment]
Sorry for not being clear.

I'm aware that WS server does not support redis sentinel yet (thats a thing I would like to work on).
My question was regarding the redis sentinel broadcast adapter, it is merged into 0.5.0-dev.

My question is what is planned with the branch 0.5.0-dev branch as 0.6.0 is already released.

**Pull request merged**